### PR TITLE
added custom Maximum Contracts per Type

### PR DESCRIPTION
**Pull request by emwedk**
take it or leave it :)

### CODE REVIEW

File: xml/defaultConfig.xml [hunk 1]
```diff
@@ -112,4 +112,23 @@
...
+        <!-- <sowMission></sowMission> -->
+        <harvestMission>12</harvestMission>
+        <!-- <hoeMission></hoeMission> -->
+        <!-- <weedMission></weedMission> -->
+        <!-- <herbicideMission></herbicideMission> -->
+        <!-- <fertilizeMission></fertilizeMission> -->
+        <!-- <mowMission></mowMission> -->
+        <!-- <tedderMission></tedderMission> -->
+        <!-- <stonePickMission></stonePickMission> -->
+        <deadwoodMission>1</deadwoodMission>
```
Comment (GMNGjoy) [inline xml/defaultConfig.xml:130]: would `<deadwoodMission>0</deadwoodMission>` also work for a player to remove those type of missions from getting generated?

File: xml/defaultConfig.xml [hunk 2]
```diff
@@ -112,4 +112,23 @@
...
         <!-- <destructibleRockMission></destructibleRockMission> -->
     </customRewards>
+
+    <customMaxType>
+        <!-- <baleMission></baleMission> -->
+        <!-- <baleWrapMission></baleWrapMission> -->
+        <!-- <plowMission></plowMission> -->
+        <!-- <cultivateMission></cultivateMission>  -->
+        <!-- <sowMission></sowMission> -->
+        <harvestMission>12</harvestMission>
```
Comment (GMNGjoy) [inline xml/defaultConfig.xml:122]: The values in the default config will get applied to every player who installs the mod - I'd prefer slightly more sane values here for the defaults... I'm fine with adding an override for harvest missions to something like 8-10, but not the default overrides to "lower" the default values for `deadwoodMission`, `treeTransportMissions`, or `destructibleRockMission` as many players love those type of missions. 

If you're in the code, we'd also need documentation added to the README.md file regarding this new configuration setting.

**Review by GMNGjoy — CHANGES_REQUESTED** [review]
First off - a humble thank you for taking the time to submit a proper PR with (hopefully) tested code! You're the first to submit something like this without me asking for it first!


This looks like a solid change, and an interesting idea. If you're willing, I'd suggest to take it a step further - and support a `0` value in the max num per type, which should allow the player to omit certain types of missions from ever being generated. 

"Don't like `destructibleRockMission`, set the `customMaxPerType` to `0`!"

but unfortunately the PR currently has a conflict with the fix i just pushed for #3. Resolve the conflict, and once I can test it - i'll add the change in the next release!

**Comment by emwedk** [comment]
i changed the code to your prefs - i just put my prefs as placeholders initially.

since i'm not a pro at lua, i'll just go ahead and assume that a value of 0 equals "false" and therefore if 0 is set in xml, it'll default to .config.maxContractsPerType.

i'll give it a test what an actual integer of 0 would do as you hypothesised

**Comment by emwedk** [comment]
testing with 
`<harvestMission>0</harvestMission>` and `<harvestMission>-1</harvestMission>` and `<harvestMission>notint</harvestMission>`
in xml 
with
`missionType.data.maxNumInstances = ContractBoost.config.customMaxPerType[typeName] == nil and ContractBoost.config.maxContractsPerType or ContractBoost.config.customMaxPerType[typeName]`
in missionBalance.lua
yielded that it's possible to stop a type of a mission to show with 0 or -1, and goes to default with a non-integer.

i've also gone ahead and set the max that you used in the configloader
and by finding that 0 isn't falsy i were able to optimize the code to a simple line

furthermore i added some explanations in the readme - change what you don't like.
i fetched the game defaults with debugging.

**Review by GMNGjoy — APPROVED** [review]
Looks good, merging it in.

**Pull request merged**